src/js/libcs/OpDrawing.js

```diff
@@ -520,168 +534,131 @@ eval_helper.drawconic = function(conicMatrix, modifs) {
+                    }
+                } else { // !vert => yFlat for horizontal tangent to oval
+                    y = flat;
+                    if (0 <= y && y <= csh) {
+                        x = -0.5 * (c11 * y + c10) / c20;
+                        if (0 <= x && x <= csw) points.push(mkp(x, y, t));
+                    }
+                }
+            }
         }
         return true;
     }
 
-    if (!(doBoundary(tl, bl, yLeft, 0, 0, 1, 1, csh) &&
-            doBoundary(bl, br, xBottom, 0, csh, 0, 1, csw) &&
-            doBoundary(tr, br, yRight, csw, 0, 1, -1, csh) &&
-            doBoundary(tl, tr, xTop, 0, 0, 0, -1, csw)))
+    if (!(doBoundary(yLeft, 0, true, +1, csh) &&
+            doBoundary(xBottom, csh, false, +1, csw) &&
+            doBoundary(yRight, csw, true, -1, csh) &&
+            doBoundary(xTop, 0, false, -1, csw)))
         return;
 
-    function segmentEllipse() {
-        if (discr <= 0)
-            return false;
-
-        var initialLength = boundary.length;
-
-        // Compute the roots of the y discriminant
-        // for points with vertical tangents
-        sol = solveRealQuadratic(k00, -2 * k10, k20);
-        if (sol) {
-            for (i = 0; i < 2; ++i)
-                if (sol[i] >= 0 && sol[i] <= csw)
-                    boundary.push(
-                        mkp(sol[i], -0.5 * (c11 * sol[i] + c01) / c02));
+    csctx.beginPath();
+    var next;
+    var n = points.length;
+    if (n < 2) return; // Nothing to draw
+    var i = n - 1;
+    var previous = points[i]; // Wrap-around
+    var previousBegin = null;
+    do {
+        next = points[i];
+        if (next.t === "begin") {
+            previousBegin = next;
+            break;
         }
-
-        // Compute the roots of the x discriminant
-        // for points with horizontal tangents
-        sol = solveRealQuadratic(k00, -2 * k01, k02);
-        if (sol) {
-            for (i = 0; i < 2; ++i)
-                if (sol[i] >= 0 && sol[i] <= csh)
-                    boundary.push(
-                        mkp(-0.5 * (c11 * sol[i] + c10) / c20, sol[i]));
+    } while (--i > 0);
+    if (!(k00 < 0 && previous.t === "end")) csctx.moveTo(previous.x, previous.y);
+    for (i = 0; i < n; ++i) {
+        next = points[i];
+        switch (next.t) {
+            case "begin":
+                previousBegin = next;
+                if (k00 < 0) csctx.moveTo(next.x, next.y);
+                else drawArc(previous, next);
+                break;
+            case "corner":
+                if (df !== "D") csctx.lineTo(next.x, next.y);
+                break;
+            case "end":
+                if (df !== "D") csctx.lineTo(next.x, next.y);
+                else csctx.moveTo(next.x, next.y);
+                if (k00 < 0) drawArc(next, previousBegin);
+                break;
+            case "split":
+                drawArc(previous, next);
+                break;
         }
```

Comment:
The code now draws an arc from the end to previous begin only when hyperbola has its center within the clipping boundaries and there are two paths. In the rest of the situations the edge segments are connected by arc from their end to their beginning into one single path. Only while filling are these segments actually a part of the path.